client/state/domains/nameservers/reducer.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DOMAIN_NAMESERVERS_FETCH,
+	DOMAIN_NAMESERVERS_FETCH_FAILURE,
+	DOMAIN_NAMESERVERS_RECEIVE,
+} from 'calypso/state/action-types';
+import { keyedReducer } from 'calypso/state/utils';
+import initialDomainState from './initial';
+
+/**
+ * Updates name servers entry for given domain.
+ *
+ * @param {object} [state] Current state.
+ * @param {object} [data] Domain name servers data.
+ * @returns {object} New state
+ */
+function updateState( state, data ) {
+	return Object.assign( {}, state, {
+		...initialDomainState,
```

Comment:
Instead of always spreading `initialDomainState` just to be sure all values are there, can `initialDefaultState` be simply a default value of `state` in the parent reducer?
```js
keyedReducer( 'domainName' ( state = initialDomainState, action ) => {
  switch ( action.type ) {
    case DOMAIN_NAMESERVERS_FETCH:
      return updateState( state, { isFetching: true, error: false } );
  }
} );
```

This should also ensure that everything that we need is always there, and it's a common Redux pattern.

Reply:
Good call! Now that we're using `keyedReducer` we actually don't need the `updateState` helper at all. I've removed it and got rid of `updateState` completely in https://github.com/Automattic/wp-calypso/pull/47272/commits/e12275ef93193e7c1d7fb2910c97aec59c04552a.

Reply:
Awesome! :+1: